### DISCUSSION TITLE
Is there a way to hook onto a callback in useWaitForTransaction?

### DISCUSSION THREAD

**Discussion by terrytjw** [wevm/wagmi#1563] [Q&A] [answered]
I'm trying to trigger a [react-hot-toast](react-hot-toast.com) popup while waiting for the transaction to be successful. Is there something like an `onLoading` in the config [here](https://wagmi.sh/react/hooks/useWaitForTransaction#configuration) for me to hook onto to run some kind of function? Or are there better ways to do it? Please advice!

**Comment by tmm** [comment, marked as answer]
How are you getting the `hash` to pass into `useWaitForTransaction`? If it's from `useContractWrite` or `useSendTransaction`, you can hook into `onSuccess` there.

**Reply by terrytjw** [reply]
Works for me. Thanks!
